lib/api/repository/login_repository.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter_survey/api/request/login_request.dart';
+import 'package:flutter_survey/api/service/login_service.dart';
+import 'package:flutter_survey/model/login_model.dart';
+import 'package:flutter_survey/api/exception/network_exceptions.dart';
+import 'package:flutter_survey/usecases/login_use_case.dart';
+import 'package:injectable/injectable.dart';
+import 'package:flutter_survey/env.dart';
+
+abstract class LoginRepository {
+  Future<LoginModel> doLogin({
+    required LoginInput input,
```

Comment:
I think the Repository layer should NOT receive the `input` model. it should receive separate `email` and `password` params instead 💭 The thing is passing a single model `LoginInput` could be so clear as the API parameters.